tools/src/main/scala/caliban/tools/SchemaWriter.scala

```diff
@@ -19,13 +19,27 @@ object SchemaWriter {
     scalarMappings: Option[Map[String, String]],
     isEffectTypeAbstract: Boolean = false,
     preserveInputNames: Boolean = false,
-    addDerives: Boolean = false
+    addDerives: Boolean = false,
+    zioEnv: Option[String] = None
   ): String = {
-    val derivesSchema: String =
-      if (addDerives) " derives caliban.schema.Schema.SemiAuto" else ""
 
-    val derivesSchemaAndArgBuilder: String =
-      if (addDerives) " derives caliban.schema.Schema.SemiAuto, caliban.schema.ArgBuilder" else ""
+    val (derivesSchema, derivesSchemaAndArgBuilder, derivesEnvSchema, derivesOnQuery) = (addDerives, zioEnv) match {
+      case (false, _)                                        => ("", "", "", "")
+      case (true, Some(env)) if !env.equalsIgnoreCase("Any") =>
+        (
+          " derives caliban.schema.Schema.SemiAuto",
+          " derives caliban.schema.Schema.SemiAuto, caliban.schema.ArgBuilder",
+          s"object EnvSchema extends caliban.schema.GenericSchema[${safeName(env)}]\n\n",
```

Comment:
Prefer extending `SchemaDerivation` instead of `GenericSchema` for Scala 3. I had issues with extending `GenericSchema` in the past